frontend/src/pages/connectionTypes/manage/ConnectionTypeDataFieldModal.tsx

```diff
@@ -209,45 +251,7 @@ export const ConnectionTypeDataFieldModal: React.FC<Props> = ({
           </FormHelperText>
         </FormGroup>
         <FormGroup fieldId="fieldType" label="Type" isRequired data-testid="field-type-select">
-          <Select
-            id="fieldType"
-            isOpen={isTypeSelectOpen}
-            shouldFocusToggleOnSelect
-            selected={fieldType}
-            onSelect={(_e, selection) => {
-              if (isConnectionTypeFieldType(selection)) {
-                setPropertiesValid(true);
-                setProperties({});
-                setFieldType(selection);
-                setIsTypeSelectOpen(false);
-              }
-            }}
-            onOpenChange={(open) => setIsTypeSelectOpen(open)}
-            toggle={(toggleRef) => (
-              <MenuToggle
-                ref={toggleRef}
-                id="type-select"
-                isFullWidth
-                onClick={() => {
-                  setIsTypeSelectOpen((open) => !open);
-                }}
-                isExpanded={isTypeSelectOpen}
-              >
-                {fieldType ? fieldTypeToString(fieldType) : ''}
-              </MenuToggle>
-            )}
-          >
-            <SelectList>
-              {connectionTypeDataFields
-                .map((value) => ({ label: fieldTypeToString(value), value }))
-                .toSorted((a, b) => a.label.localeCompare(b.label))
-                .map(({ value, label }) => (
-                  <SelectOption key={value} value={value} data-testid={`field-${value}-select`}>
-                    {label}
-                  </SelectOption>
-                ))}
-            </SelectList>
-          </Select>
+          {typeSelect}
```

Comment:
This change is necessary because the `eslint` rule is flawed. It doesn't allow any children of the `Modal` to use an `isOpen` parameter (like to the `Select` component here.